checker_test.cpp

```diff
@@ -0,0 +1,43 @@
+#include <iostream>
+#include <assert.h>
+#include "checker_test.h"
+
+using namespace std;
+
+void temperatureCheck_test(){
+  cout<< "Temperature Check tests: START" <<endl;
+  assert(temperatureCheck(minTemperature) == true);
+  assert(temperatureCheck(maxTemperature) == true);
+  assert(temperatureCheck(minTemperature - 0.1) == false);
+  assert(temperatureCheck(maxTemperature + 0.1) == false);
+  cout<< "Temperature Check tests: END, success!" <<endl;
+}
+
+void stateOfChargeCheck_test(){
+  cout<< "State of charge Check tests: START" <<endl;
+  assert(stateOfChargeCheck(minSOC) == true);
+  assert(stateOfChargeCheck(maxSOC) == true);
+  assert(stateOfChargeCheck(minSOC - 0.1) == false);
+  assert(stateOfChargeCheck(maxSOC + 0.1) == false);
+  cout<< "State of charge Check tests: END, success!" <<endl;
+}
+
+void chargeRateCheck_test(){
+  cout<< "Charge Rate Check tests: START" <<endl;
+  assert(chargeRateCheck(maxChargeRate) == true);
+  assert(chargeRateCheck(maxChargeRate - 0.1) == true);
+  assert(chargeRateCheck(maxChargeRate + 0.1) == false);
+  cout<< "Charge Rate Check tests: END, success!" <<endl;
+}
+
+void batteryOk_test(){
```

Comment:
Nice approach to test the parts (individual parameters) and then the integrated `batteryOk` function.
The chargeRate parameter is not tested in here. It means, if someone removes the `chargeRateCheck` from `batteryOk`, the tests will still pass.

Reply:
I think I got the point, added battery scenarios per each variable flaw. Thanks again for the feedback!